Compile/src/Producers/PrefiringWeightProducer.cc

```diff
@@ -1,162 +1,338 @@
+        if (!missingInputMuon_) {
+            for (KMuons::iterator muon = theMuons->begin(); muon != theMuons->end(); ++muon) {
+                double pt = muon->p4.Pt();
+                double phi = muon->p4.Phi();
+                double eta = muon->p4.Eta();
+                // Remove crappy tracker muons which would not have prefired the L1 trigger
+                if (pt < 5 || !muon->idLoose())
+                    continue;
+                double prefiringprob_mu = getPrefiringRateMuon(eta, phi, pt, fluct);
+                nonPrefiringProbaMuon[fluct] *= (1. - prefiringprob_mu);
             }
-            //Last case: if overlapping photons have a non prefiring rate smaller than the jet, don't consider the jet in the event weight, and do nothing.
         }
     }
+    //Move global prefire weights, as well as those for muons, photons, and jets, to the event
+    //Up and Down are switched, since we use the inverse and apply it to data.
     product.m_optionalWeights["prefiringWeight"] =  1./nonPrefiringProba[0];
-    product.m_optionalWeights["prefiringWeightUp"] =  1./nonPrefiringProba[1];
-    product.m_optionalWeights["prefiringWeightDown"] =  1./nonPrefiringProba[2];
-  
+    product.m_optionalWeights["prefiringWeightUp"] =  1./nonPrefiringProba[2];
+    product.m_optionalWeights["prefiringWeightDown"] =  1./nonPrefiringProba[1];
+
+    product.m_optionalWeights["prefiringWeightECAL"] =  1./nonPrefiringProbaECAL[0];
+    product.m_optionalWeights["prefiringWeightECALUp"] =  1./nonPrefiringProbaECAL[2];
+    product.m_optionalWeights["prefiringWeightECALDown"] =  1./nonPrefiringProbaECAL[1];
+
+    product.m_optionalWeights["prefiringWeightMuon"] =  1./nonPrefiringProbaMuon[0];
+    product.m_optionalWeights["prefiringWeightMuonUp"] =  1./nonPrefiringProbaMuon[2];
+    product.m_optionalWeights["prefiringWeightMuonDown"] =  1./nonPrefiringProbaMuon[1];
+    product.m_optionalWeights["prefiringWeightMuonUpStat"] =  1./nonPrefiringProbaMuon[4];
+    product.m_optionalWeights["prefiringWeightMuonDownStat"] =  1./nonPrefiringProbaMuon[3];
+    product.m_optionalWeights["prefiringWeightMuonUpSyst"] =  1./nonPrefiringProbaMuon[6];
+    product.m_optionalWeights["prefiringWeightMuonDownSyst"] =  1./nonPrefiringProbaMuon[5];
 }
 
+double PrefiringWeightProducer::getPrefiringRateEcal(double eta,
+                                                     double pt,
+                                                     TH2F* h_prefmap,
+                                                     fluctuations fluctuation) const {
+    //Check pt is not above map overflow
+    int nbinsy = h_prefmap->GetNbinsY();
+    double maxy = h_prefmap->GetYaxis()->GetBinLowEdge(nbinsy + 1);
+    if (pt >= maxy)
+        pt = maxy - 0.01;
+    int thebin = h_prefmap->FindBin(eta, pt);
+
+    double prefrate = h_prefmap->GetBinContent(thebin);
+
+    double statuncty = h_prefmap->GetBinError(thebin);
```

Comment:
Wasn't this the combined error?
Or am I confusing something here?

Reply:
This was for Muon Scale factors. Here, we only have sepearte errors for the Muon Prefiring. For the ECAL there is only the combined error available.